### PR TITLE
Create luhnCheck.md

### PR DESCRIPTION
**Pull request by kriadmin**
## Description
Create luhnCheck to validate a variety of identification numbers, such as credit card numbers, IMEI numbers, National Provider Identifier numbers etc.

## What does your PR belong to?
- [ ] Website
- [x] Snippets
- [ ] General / Things regarding the repository (like CI Integration)

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (non-breaking improvement of a snippet)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have checked that the changes are working properly
- [x] I have checked that there isn't any PR doing the same
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: snippets/luhnCheck.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+### luhnCheck
+
+Implementation of the [Luhn Algorithm](https://en.wikipedia.org/wiki/Luhn_algorithm) used to validate a variety of identification numbers, such as credit card numbers, IMEI numbers, National Provider Identifier numbers etc.
+Works with numbers and strings alike
+``` js 
+const luhnCheck = num => {
```
Comment (Chalarangelo) [inline snippets/luhnCheck.md]: Maybe rename to `validateChecksum`?

File: snippets/luhnCheck.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+const luhnCheck = num => { 
+    let arr = (num+'').split('').reverse()
+    let lastDigit = arr.splice(0,1)
+    let sum = arr.reduce((acc,val,i) => {
+	    val = parseInt(val)
+    	if(i%2!==0) return acc + val
+    	else{
+		    val = (val * 2)%9 || 9
+		    return acc+ val }
+        },0)
```
Comment (Chalarangelo) [inline snippets/luhnCheck.md]: This whole `reduce` could be one-lined like so:
```js
(acc,val,i) => i%2!==0 ? acc + parseInt(val) : acc + (parseInt(val) * 2)%9 || 9;
``

**Comment by kriadmin** [comment]
@Chalarangelo I updated the changes as suggested by you. Not sure about the name 😕

**Comment by Chalarangelo** [comment]
Changes look solid. I need opinions on the name @kingdavidmartins @fejes713 and anyone else who is currently online.

**Comment by kriadmin** [comment]
@Chalarangelo The code has stopped working after the last update!

File: snippets/luhnCheck.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+### luhnCheck
+
+Implementation of the [Luhn Algorithm](https://en.wikipedia.org/wiki/Luhn_algorithm) used to validate a variety of identification numbers, such as credit card numbers, IMEI numbers, National Provider Identifier numbers etc.
+Works with numbers and strings alike
+``` js 
+const luhnCheck = num => { 
+    let arr = (num+'').split('').reverse()
+    let lastDigit = arr.splice(0,1)
+    let sum = arr.reduce((acc,val,i) => i%2!==0 ? acc + parseInt(val) : acc + (parseInt(val) * 2)%9 || 9,0)
```
Comment (Chalarangelo) [inline snippets/luhnCheck.md]: Should be `(acc,val,i) => i%2!==0 ? acc + parseInt(val) : acc + ((parseInt(val) * 2)%9 || 9);`

Comment (kriadmin) [reply snippets/luhnCheck.md]: @Chalarangelo No it works fine now. I mistakingly   `(acc,val,i) => ` twice . I shuld have checked the code before commenting.

**Comment by fejes713** [comment]
Wiki says:
>It is a simple checksum formula used to validate a variety of identification numbers

@Chalarangelo, So I guess`validateChecksum` is good name for it.

**Comment by kriadmin** [comment]
@Chalarangelo `LuhnCheckCard` `validCreditCard` `luhn`  `luhnValid` `luhnValidate ` are some other names you can consider before taking the final decision

**Comment by Chalarangelo** [comment]
@kriadmin As far as I can tell you can validate checksums other than credit cards, correct?

**Comment by kriadmin** [comment]
@Chalarangelo I don't know I tested it only with CC numbers 😕

**Comment by skatcat31** [comment]
The luhn algorithm is used on many things. I like keeping the name luhn related because it is NOT valid on other checksums nor is it cryptographically secure. It is merely to prevent against errors in many places, like CC numbers, which makes it useful enough

**Comment by Chalarangelo** [comment]
@skatcat31 Thanks for the input, really appreciated! So we are keeping the name as-is. I will make some minor changes to the description/explanation and merge afterwards!

**Pull request merged**